--- libev.spec.orig	2015-10-26 21:10:49.489018944 +0900
+++ libev.spec	2015-10-26 21:15:16.492020449 +0900
@@ -56,6 +56,7 @@ This package contains the source code fo
 autoreconf -i
 
 %build
+autoreconf --force --install
 %configure --disable-static --with-pic
 make %{?_smp_mflags}
 
@@ -75,8 +76,7 @@ install -p -m 0644 Changes ev.pod LICENS
 %postun -p /sbin/ldconfig
 
 %files
-%license LICENSE
-%doc Changes README
+%doc LICENSE Changes README
 %{_libdir}/%{name}.so.4
 %{_libdir}/%{name}.so.4.0.0
 
