--- /vagrant/httpd.spec.24	2015-10-25 07:44:44.046869700 +0900
+++ httpd.spec	2015-10-25 10:05:26.103940784 +0900
@@ -3,22 +3,23 @@
 %define suexec_caller apache
 %define mmn 20120211
 %define mmnisa %{mmn}%{__isa_name}%{__isa_bits}
-%define vstring %(source /etc/os-release; echo ${REDHAT_SUPPORT_PRODUCT})
+%define vstring CentOS
 
 Summary: Apache HTTP Server
 Name: httpd
 Version: 2.4.17
-Release: 2%{?dist}
+Release: 3%{?dist}
 URL: http://httpd.apache.org/
 Source0: http://www.apache.org/dist/httpd/httpd-%{version}.tar.bz2
-Source1: index.html
+Source1: centos_index.html
 Source2: httpd.logrotate
 Source3: httpd.sysconf
-Source4: httpd-ssl-pass-dialog
-Source5: httpd.tmpfiles
-Source6: httpd.service
-Source7: action-graceful.sh
-Source8: action-configtest.sh
+#Source4: httpd-ssl-pass-dialog
+#Source5: httpd.tmpfiles
+#Source6: httpd.service
+#Source7: action-graceful.sh
+#Source8: action-configtest.sh
+Source9: httpd.init
 Source10: httpd.conf
 Source11: 00-base.conf
 Source12: 00-mpm.conf
@@ -33,16 +34,17 @@ Source20: userdir.conf
 Source21: ssl.conf
 Source22: welcome.conf
 Source23: manual.conf
-Source24: 00-systemd.conf
+#Source24: 00-systemd.conf
 Source25: 01-session.conf
-Source26: 10-listen443.conf
-Source27: httpd.socket
+#Source26: 10-listen443.conf
+#Source27: httpd.socket
 Source28: 00-optional.conf
 # Documentation
 Source30: README.confd
 Source31: README.confmod
-Source40: htcacheclean.service
+#Source40: htcacheclean.service
 Source41: htcacheclean.sysconf
+Source42: htcacheclean.init
 # build/scripts patches
 Patch1: httpd-2.4.1-apctl.patch
 Patch2: httpd-2.4.9-apxs.patch
@@ -73,18 +75,19 @@ BuildRoot: %{_tmppath}/%{name}-%{version
 BuildRequires: autoconf, perl, pkgconfig, findutils, xmlto
 BuildRequires: zlib-devel, libselinux-devel, lua-devel
 BuildRequires: apr-devel >= 1.5.0, apr-util-devel >= 1.5.0, pcre-devel >= 5.0
-BuildRequires: systemd-devel, libnghttp2-devel
-Requires: /etc/mime.types, system-logos-httpd
+BuildRequires: libnghttp2-devel
+Requires: initscripts >= 8.36, /etc/mime.types, system-logos >= 7.92.1-1
 Obsoletes: httpd-suexec
+Requires(pre): /usr/sbin/useradd, httpd-filesystem
+Requires(post): chkconfig
 Provides: webserver
 Provides: mod_dav = %{version}-%{release}, httpd-suexec = %{version}-%{release}
 Provides: httpd-mmn = %{mmn}, httpd-mmn = %{mmnisa}
-Requires: httpd-tools = %{version}-%{release}
+Obsoletes: apache, secureweb, mod_dav, mod_gzip, stronghold-apache
+Obsoletes: stronghold-htdocs, mod_put, mod_roaming
+Conflicts: pcre < 4.0
+Requires: httpd-tools = %{version}-%{release}, apr-util
 Requires: httpd-filesystem = %{version}-%{release}
-Requires(pre): httpd-filesystem
-Requires(preun): systemd-units
-Requires(postun): systemd-units
-Requires(post): systemd-units
 Conflicts: apr < 1.5.0-1
 
 %description
@@ -289,11 +292,11 @@ rm -rf $RPM_BUILD_ROOT
 make DESTDIR=$RPM_BUILD_ROOT install
 
 # Install systemd service files
-mkdir -p $RPM_BUILD_ROOT%{_unitdir}
-for s in httpd.service htcacheclean.service httpd.socket; do
-  install -p -m 644 $RPM_SOURCE_DIR/${s} \
-                    $RPM_BUILD_ROOT%{_unitdir}/${s}
-done
+#mkdir -p $RPM_BUILD_ROOT%{_unitdir}
+#for s in httpd.service htcacheclean.service httpd.socket; do
+#  install -p -m 644 $RPM_SOURCE_DIR/${s} \
+#                    $RPM_BUILD_ROOT%{_unitdir}/${s}
+#done
 
 # install conf file/directory
 mkdir $RPM_BUILD_ROOT%{_sysconfdir}/httpd/conf.d \
@@ -304,7 +307,7 @@ install -m 644 $RPM_SOURCE_DIR/README.co
     $RPM_BUILD_ROOT%{_sysconfdir}/httpd/conf.modules.d/README
 for f in 00-base.conf 00-mpm.conf 00-lua.conf 01-cgi.conf 00-dav.conf \
          00-proxy.conf 00-ssl.conf 01-ldap.conf 00-proxyhtml.conf \
-         01-ldap.conf 00-systemd.conf 01-session.conf 00-optional.conf; do
+         01-ldap.conf 01-session.conf 00-optional.conf; do
   install -m 644 -p $RPM_SOURCE_DIR/$f \
         $RPM_BUILD_ROOT%{_sysconfdir}/httpd/conf.modules.d/$f
 done
@@ -312,11 +315,11 @@ done
 # install systemd override drop directory
 # Web application packages can drop snippets into this location if
 # they need ExecStart[pre|post].
-mkdir $RPM_BUILD_ROOT%{_unitdir}/httpd.service.d
-mkdir $RPM_BUILD_ROOT%{_unitdir}/httpd.socket.d
+#mkdir $RPM_BUILD_ROOT%{_unitdir}/httpd.service.d
+#mkdir $RPM_BUILD_ROOT%{_unitdir}/httpd.socket.d
 
-install -m 644 -p $RPM_SOURCE_DIR/10-listen443.conf \
-      $RPM_BUILD_ROOT%{_unitdir}/httpd.socket.d/10-listen443.conf
+#install -m 644 -p $RPM_SOURCE_DIR/10-listen443.conf \
+#      $RPM_BUILD_ROOT%{_unitdir}/httpd.socket.d/10-listen443.conf
 
 for f in welcome.conf ssl.conf manual.conf userdir.conf; do
   install -m 644 -p $RPM_SOURCE_DIR/$f \
@@ -343,13 +346,13 @@ for s in httpd htcacheclean; do
 done
 
 # tmpfiles.d configuration
-mkdir -p $RPM_BUILD_ROOT%{_prefix}/lib/tmpfiles.d 
-install -m 644 -p $RPM_SOURCE_DIR/httpd.tmpfiles \
-   $RPM_BUILD_ROOT%{_prefix}/lib/tmpfiles.d/httpd.conf
+#mkdir -p $RPM_BUILD_ROOT%{_prefix}/lib/tmpfiles.d 
+#install -m 644 -p $RPM_SOURCE_DIR/httpd.tmpfiles \
+#   $RPM_BUILD_ROOT%{_prefix}/lib/tmpfiles.d/httpd.conf
 
 # Other directories
 mkdir -p $RPM_BUILD_ROOT%{_localstatedir}/lib/dav \
-         $RPM_BUILD_ROOT/run/httpd/htcacheclean
+         $RPM_BUILD_ROOT%{_localstatedir}/run/httpd/htcacheclean
 
 # Substitute in defaults which are usually done (badly) by "make install"
 sed -i \
@@ -378,9 +381,9 @@ cat > $RPM_BUILD_ROOT%{_rpmconfigdir}/ma
 EOF
 
 # Handle contentdir
-mkdir $RPM_BUILD_ROOT%{contentdir}/noindex
-install -m 644 -p $RPM_SOURCE_DIR/index.html \
-        $RPM_BUILD_ROOT%{contentdir}/noindex/index.html
+mkdir $RPM_BUILD_ROOT%{contentdir}/html
+install -m 644 -p $RPM_SOURCE_DIR/centos_index.html \
+        $RPM_BUILD_ROOT%{contentdir}/error/noindex.html
 rm -rf %{contentdir}/htdocs
 
 # remove manual sources
@@ -403,25 +406,32 @@ rm -v $RPM_BUILD_ROOT%{docroot}/html/*.h
       $RPM_BUILD_ROOT%{docroot}/cgi-bin/*
 
 # Symlink for the powered-by-$DISTRO image:
-ln -s ../../pixmaps/poweredby.png \
+ln -s ../../../..%{_datadir}/pixmaps/poweredby.png \
         $RPM_BUILD_ROOT%{contentdir}/icons/poweredby.png
 
 # symlinks for /etc/httpd
 ln -s ../..%{_localstatedir}/log/httpd $RPM_BUILD_ROOT/etc/httpd/logs
-ln -s /run/httpd $RPM_BUILD_ROOT/etc/httpd/run
+ln -s ../..%{_localstatedir}/run/httpd $RPM_BUILD_ROOT/etc/httpd/run
 ln -s ../..%{_libdir}/httpd/modules $RPM_BUILD_ROOT/etc/httpd/modules
 
 # install http-ssl-pass-dialog
-mkdir -p $RPM_BUILD_ROOT%{_libexecdir}
-install -m755 $RPM_SOURCE_DIR/httpd-ssl-pass-dialog \
-	$RPM_BUILD_ROOT%{_libexecdir}/httpd-ssl-pass-dialog
+#mkdir -p $RPM_BUILD_ROOT%{_libexecdir}
+#install -m755 $RPM_SOURCE_DIR/httpd-ssl-pass-dialog \
+#	$RPM_BUILD_ROOT%{_libexecdir}/httpd-ssl-pass-dialog
 
 # Install action scripts
-mkdir -p $RPM_BUILD_ROOT%{_libexecdir}/initscripts/legacy-actions/httpd
-for f in graceful configtest; do
-    install -p -m 755 $RPM_SOURCE_DIR/action-${f}.sh \
-            $RPM_BUILD_ROOT%{_libexecdir}/initscripts/legacy-actions/httpd/${f}
-done
+#mkdir -p $RPM_BUILD_ROOT%{_libexecdir}/initscripts/legacy-actions/httpd
+#for f in graceful configtest; do
+#    install -p -m 755 $RPM_SOURCE_DIR/action-${f}.sh \
+#            $RPM_BUILD_ROOT%{_libexecdir}/initscripts/legacy-actions/httpd/${f}
+#done
+
+# install SYSV init stuff
+mkdir -p $RPM_BUILD_ROOT/etc/rc.d/init.d
+install -m755 $RPM_SOURCE_DIR/httpd.init \
+	$RPM_BUILD_ROOT/etc/rc.d/init.d/httpd
+install -m755 $RPM_SOURCE_DIR/htcacheclean.init \
+	$RPM_BUILD_ROOT/etc/rc.d/init.d/htcacheclean
 
 # Install logrotate config
 mkdir -p $RPM_BUILD_ROOT/etc/logrotate.d
@@ -434,7 +444,7 @@ sed -e "s|/usr/local/apache2/conf/httpd.
     -e "s|/usr/local/apache2/conf/magic|/etc/httpd/conf/magic|" \
     -e "s|/usr/local/apache2/logs/error_log|/var/log/httpd/error_log|" \
     -e "s|/usr/local/apache2/logs/access_log|/var/log/httpd/access_log|" \
-    -e "s|/usr/local/apache2/logs/httpd.pid|/run/httpd/httpd.pid|" \
+    -e "s|/usr/local/apache2/logs/httpd.pid|/var/run/httpd/httpd.pid|" \
     -e "s|/usr/local/apache2|/etc/httpd|" < docs/man/httpd.8 \
   > $RPM_BUILD_ROOT%{_mandir}/man8/httpd.8
 
@@ -460,6 +470,9 @@ rm -vf \
 
 rm -rf $RPM_BUILD_ROOT/etc/httpd/conf/{original,extra}
 
+# Make suexec a+rw so it can be stripped.  %%files lists real permissions
+chmod 755 $RPM_BUILD_ROOT%{_sbindir}/suexec
+
 %pre
 # Add the "apache" user
 /usr/sbin/useradd -c "Apache" -u 48 \
@@ -471,28 +484,35 @@ rm -rf $RPM_BUILD_ROOT/etc/httpd/conf/{o
 	-s /sbin/nologin -r -d %{contentdir} apache 2> /dev/null || :
 
 %post
-%systemd_post httpd.service htcacheclean.service httpd.socket
+# Register the httpd service
+/sbin/chkconfig --add httpd
+/sbin/chkconfig --add htcacheclean
 
 %preun
-%systemd_preun httpd.service htcacheclean.service httpd.socket
-
-%postun
-%systemd_postun
-
-# Trigger for conversion from SysV, per guidelines at:
-# https://fedoraproject.org/wiki/Packaging:ScriptletSnippets#Systemd
-%triggerun -- httpd < 2.2.21-5
-# Save the current service runlevel info
-# User must manually run systemd-sysv-convert --apply httpd
-# to migrate them to systemd targets
-/usr/bin/systemd-sysv-convert --save httpd.service >/dev/null 2>&1 ||:
+if [ $1 = 0 ]; then
+	/sbin/service httpd stop > /dev/null 2>&1
+	/sbin/chkconfig --del httpd
+	/sbin/service htcacheclean stop > /dev/null 2>&1
+	/sbin/chkconfig --del htcacheclean
+fi
 
-# Run these because the SysV package being removed won't do them
-/sbin/chkconfig --del httpd >/dev/null 2>&1 || :
+#%postun
+#%systemd_postun
+#
+## Trigger for conversion from SysV, per guidelines at:
+## https://fedoraproject.org/wiki/Packaging:ScriptletSnippets#Systemd
+#%triggerun -- httpd < 2.2.21-5
+## Save the current service runlevel info
+## User must manually run systemd-sysv-convert --apply httpd
+## to migrate them to systemd targets
+#/usr/bin/systemd-sysv-convert --save httpd.service >/dev/null 2>&1 ||:
+#
+## Run these because the SysV package being removed won't do them
+#/sbin/chkconfig --del httpd >/dev/null 2>&1 || :
 
 %posttrans
 test -f /etc/sysconfig/httpd-disable-posttrans || \
-  /bin/systemctl try-restart httpd.service htcacheclean.service >/dev/null 2>&1 || :
+ /sbin/service httpd condrestart >/dev/null 2>&1 || :
 
 %define sslcert %{_sysconfdir}/pki/tls/certs/localhost.crt
 %define sslkey %{_sysconfdir}/pki/tls/private/localhost.key
@@ -556,6 +576,8 @@ rm -rf $RPM_BUILD_ROOT
 %config(noreplace) %{_sysconfdir}/httpd/conf/magic
 
 %config(noreplace) %{_sysconfdir}/logrotate.d/httpd
+%{_sysconfdir}/rc.d/init.d/httpd
+%{_sysconfdir}/rc.d/init.d/htcacheclean
 
 %config(noreplace) %{_sysconfdir}/httpd/conf.d/*.conf
 %exclude %{_sysconfdir}/httpd/conf.d/ssl.conf
@@ -570,10 +592,10 @@ rm -rf $RPM_BUILD_ROOT
 %exclude %{_sysconfdir}/httpd/conf.modules.d/01-session.conf
 
 %config(noreplace) %{_sysconfdir}/sysconfig/ht*
-%{_prefix}/lib/tmpfiles.d/httpd.conf
+#%{_prefix}/lib/tmpfiles.d/httpd.conf
 
-%dir %{_libexecdir}/initscripts/legacy-actions/httpd
-%{_libexecdir}/initscripts/legacy-actions/httpd/*
+#%dir %{_libexecdir}/initscripts/legacy-actions/httpd
+#%{_libexecdir}/initscripts/legacy-actions/httpd/*
 
 %{_sbindir}/ht*
 %{_sbindir}/fcgistarter
@@ -593,15 +615,14 @@ rm -rf $RPM_BUILD_ROOT
 
 %dir %{contentdir}/error
 %dir %{contentdir}/error/include
-%dir %{contentdir}/noindex
 %{contentdir}/icons/*
 %{contentdir}/error/README
 %{contentdir}/error/*.var
 %{contentdir}/error/include/*.html
-%{contentdir}/noindex/index.html
+%{contentdir}/error/noindex.html
 
-%attr(0710,root,apache) %dir /run/httpd
-%attr(0700,apache,apache) %dir /run/httpd/htcacheclean
+%attr(0710,root,apache) %dir %{_localstatedir}/run/httpd
+%attr(0700,apache,apache) %dir %{_localstatedir}/run/httpd/htcacheclean
 %attr(0700,root,root) %dir %{_localstatedir}/log/httpd
 %attr(0700,apache,apache) %dir %{_localstatedir}/lib/dav
 %attr(0700,apache,apache) %dir %{_localstatedir}/cache/httpd
@@ -609,10 +630,10 @@ rm -rf $RPM_BUILD_ROOT
 
 %{_mandir}/man8/*
 
-%{_unitdir}/*.service
-%{_unitdir}/*.socket
-%attr(755,root,root) %dir %{_unitdir}/httpd.service.d
-%attr(755,root,root) %dir %{_unitdir}/httpd.socket.d
+#%{_unitdir}/*.service
+#%{_unitdir}/*.socket
+#%attr(755,root,root) %dir %{_unitdir}/httpd.service.d
+#%attr(755,root,root) %dir %{_unitdir}/httpd.socket.d
 
 %files filesystem
 %dir %{_sysconfdir}/httpd
@@ -643,8 +664,8 @@ rm -rf $RPM_BUILD_ROOT
 %config(noreplace) %{_sysconfdir}/httpd/conf.modules.d/00-ssl.conf
 %config(noreplace) %{_sysconfdir}/httpd/conf.d/ssl.conf
 %attr(0700,apache,root) %dir %{_localstatedir}/cache/httpd/ssl
-%{_libexecdir}/httpd-ssl-pass-dialog
-%{_unitdir}/httpd.socket.d/10-listen443.conf
+#%{_libexecdir}/httpd-ssl-pass-dialog
+#%{_unitdir}/httpd.socket.d/10-listen443.conf
 
 %files -n mod_proxy_html
 %defattr(-,root,root)
@@ -674,6 +695,9 @@ rm -rf $RPM_BUILD_ROOT
 %{_rpmconfigdir}/macros.d/macros.httpd
 
 %changelog
+* Sun Oct 25 2015 Kikyou Akino <bellflower@web4u.jp> 2.4.17-3
+- Rebuilt for CentOS 6.7
+
 * Wed Oct 14 2015 Jan Kaluza <jkaluza@redhat.com> - 2.4.17-2
 - rebase socket activation patch to 2.4.17
 
