--- libevent.spec.orig	2012-04-23 23:03:09.000000000 +0900
+++ libevent.spec	2015-10-18 12:34:24.522926245 +0900
@@ -1,6 +1,6 @@
 Name:           libevent
-Version:        1.4.13
-Release:        4%{?dist}
+Version:        1.4.15
+Release:        1%{?dist}
 Summary:        Abstract asynchronous event notification library
 
 Group:          System Environment/Libraries
@@ -9,9 +9,9 @@ URL:            http://monkey.org/~provo
 Source0:        http://monkey.org/~provos/libevent-%{version}-stable.tar.gz
 BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
-BuildRequires: doxygen
+BuildRequires:  libtool doxygen
 
-Patch00: libevent-1.4.13-stable-configure.patch
+Patch00:        libevent-1.4.15-stable-configure.patch
 
 %description
 The libevent API provides a mechanism to execute a callback function
@@ -55,12 +55,13 @@ This package contains the header files f
 develop programs using %{name}, you will need to install %{name}-devel.
 
 %prep
-%setup -q -n libevent-%{version}-stable
+%setup -q -n libevent-release-%{version}-stable
 
 # 477685 -  libevent-devel multilib conflict
 %patch00 -p1
 
 %build
+./autogen.sh
 %configure \
     --disable-dependency-tracking
 make %{?_smp_mflags}
@@ -136,6 +137,9 @@ rm -rf $RPM_BUILD_ROOT
 %defattr(-,root,root,0644)
 
 %changelog
+* Sun Oct 18 2015 Kikyou Akino <bellflower@web4u.jp> 1.4.15-1
+- Updated to latest stable upstream version: 1.4.15
+
 * Mon Apr 23 2012 Steve Dickson <steved@redhat.com> 1.4.13-4
 - Moved header files into there own rpm (bz 658051)
 - Added event-config.h to the new headers rpm.
